tests/components/GenKeyWarning.test.js

```diff
@@ -0,0 +1,280 @@
+
+        // Test the close button
+        await closeBtn.trigger('click');
+        await nextTick();
+        checkEventsEmitted(wrapper, 2, 1, 0);
+        expect(wrapper.emitted('close')).toStrictEqual([[], []]);
+    });
+    it('GenKeyWarning (with encrypt)', async () => {
+        // Mock translate and createAlert and the two translations used
+        vi.spyOn(translation, 'tr').mockImplementation((message, variables) => {
+            return message + variables[0].MIN_PASS_LENGTH;
+        });
+        vi.spyOn(misc, 'createAlert').mockImplementation(
+            (type, message, timeout = 0) => {
+                return message;
+            }
+        );
+        vi.spyOn(translation, 'ALERTS', 'get').mockReturnValue({
+            PASSWORD_TOO_SMALL: 'pass_too_small',
+            PASSWORD_DOESNT_MATCH: 'pass_doesnt_match',
+        });
+        const wrapper = mount(GenKeyWarning, {
+            props: {
+                showModal: true,
+                showBox: true,
+                isEncrypt: true,
+            },
+            attachTo: document.getElementById('app'),
+        });
+        checkEventsEmitted(wrapper, 0, 0, 0);
+        const encryptBox = wrapper.find('[data-testid=encryptBox]');
+        expect(encryptBox.isVisible()).toBeTruthy();
+        // Modal exist
+        checkModalExistence(wrapper, 1);
+
+        const newPassword = wrapper
+            .findComponent(Modal)
+            .findAll('[data-testid=newPasswordModal]')[0];
+        const confirmPassword = wrapper
+            .findComponent(Modal)
+            .findAll('[data-testid=confirmPasswordModal]')[0];
+        const submitBtn = wrapper
+            .findComponent(Modal)
+            .findAll('[data-testid=submitBtn]')[0];
+        const currentPassword = wrapper
+            .findComponent(Modal)
+            .findAll('[data-testid=currentPasswordModal]')[0];
+        const closeBtn = wrapper
+            .findComponent(Modal)
+            .findAll('[data-testid=closeBtn]')[0];
+        // Every input box/ button must be visible
+        expect(newPassword.isVisible()).toBeTruthy();
+        expect(confirmPassword.isVisible()).toBeTruthy();
+        expect(submitBtn.isVisible()).toBeTruthy();
+        expect(currentPassword.isVisible()).toBeTruthy();
+        expect(closeBtn.isVisible()).toBeTruthy();
+
+        // Let's try to insert a password
+        // Password match but it's too short!
+        newPassword.element.value = 'p';
+        newPassword.trigger('input');
+        await nextTick();
+        confirmPassword.element.value = 'p';
+        confirmPassword.trigger('input');
+        await nextTick();
+        await submitBtn.trigger('click');
+        await nextTick();
+        // no event should have been emitted
+        checkEventsEmitted(wrapper, 0, 0, 0);
+        expect(misc.createAlert).toHaveBeenCalled();
+        expect(misc.createAlert).toHaveReturnedWith('pass_too_small6');
```

Comment:
This `expect` seems to have a typo or a hardcoded concatenated `MIN_PASS_LENGTH`, if so, we should expect `'pass_too_small' + MIN_PASS_LENGTH` in the case that we ever change that constant, or the tests would fail.

Reply:
nice catch, review applied